### ISSUE TITLE
Pester 4 RC, PowerShell Core regression: $PSVersionTable.PSVersion change

### ISSUE THREAD

**Issue by Jaykul** [pester/Pester#864]
When I `Invoke-Pester` on Pester 4.0.6rc on PowerShell v6.0.0-beta.5 **Core** (on Windows 10) a lot of tests are failing because of version comparisons. In PowerShell Core, `$PSVersionTable.PSVersion` is a `System.Management.Automation.SemanticVersion` and so these tests all threw an error in the Describe block because they can't compare versions:

* `Describing Analyzing coverage of a DSC configuration` 
* `Describing When calling Mock on an alias` 
* `Describing When calling Mock on existing cmdlet to handle pipelined input`
* `Describing When calling Mock on cmdlet Used by Mock`
* `Describing Dot Source Test`

The error basically just says:

> PSArgumentException: Cannot process argument because the value of argument "version" is not valid. Change the value of the "version" argument and run the operation again.
> PSInvalidCastException: Cannot convert value "5.0.10586.122" to type "System.Management.Automation.SemanticVersion". Error: "Cannot process argument because the value of argument "version" is not valid. Change the value of the "version" argument and run the operation again."

Additionally, the following tests failed in the assertions, but with the same message about SemanticVersion casting ...

* `Describing Mocking Cmdlets with dynamic parameters`
    [-] Allows calls to be made with dynamic parameters (including parameter filters)
* `Describing Mocking Cmdlets with dynamic parameters in a module`
    [-] Allows calls to be made with dynamic parameters (including parameter filters)
* `Describing DynamicParam blocks in other scopes`
    [-] Properly evaluates dynamic parameters when called from another scope
    [-] Properly evaluates dynamic parameters when called from another scope when the call is from a ValidateScript block
* `Describing Mocking Get-ItemProperty`
    [-] Does not fail with NotImplementedException
* `Describing Mocking module-qualified calls`
    [-] Calls the mock properly even if the call is module-qualified

And finally, this test also failed for a totally unrelated error, but is already mentioned in #838 ☹️ 

* `Describing ConvertTo-FailureLines`
    [-] failed should produces correct message lines.

A total of 12 tests failed -- all tests pass running in PowerShell 5.1 on the same system.

**Comment by it-praktyk** [comment]
@Jaykul, thank you for your report. 

Could you test the code what you can find in the branch https://github.com/it-praktyk/Pester/tree/DevV4-1_Linux_compatibility_2?

Most of the issues are currently resolved.

**Comment by it-praktyk** [comment]
BTW, you didn't provide what version of PSCore and OS was used for your test.

**Comment by Jaykul** [comment]
Updated OP. It was beta 5 (I thought I was on b6, but wasn't) on Windows

**Comment by it-praktyk** [comment]
Currently, the pull request #925 wait for merging so I assume that we can close this issue.

**Comment by Jaykul** [comment]
Normally, one doesn't close issues **before** merging the PR.
Instead, you mention the issue as fixed in the PR, and *when* you merge it, the issues get closed.

**Comment by it-praktyk** [comment]
I'm reopening the issue. I hope that it will you feel better.

**Comment by aaronpowell** [comment]
I just hit this same issue in adding Linux support to a module of mine (https://github.com/aaronpowell/ps-nvmw/issues/35).

Digging through it I came to find that the problem was caused by this line in `Mock.ps1`:

```powershell
if ($PSVersionTable.PSVersion -ge '5.0.10586.122')
```

Changing it to

```powershell
if ('5.0.10586.122' -lt $PSVersionTable.PSVersion)
```

From the branch above has seen it working now.

**Comment by it-praktyk** [comment]
Fixed by #925
